README.md

```diff
@@ -1,2 +1,131 @@
-# termviz
-ROS visualisation on the terminal
+# TermViz - ROS visualization on the terminal
+
+TermViz is a visualizer for the Robot Operating System (ROS) framework, that runs completely on the terminal.
+This allows the visualizer to run directly on the robot (e.g., via ssh) without any configuration on an external computer.
+It can also be useful in case the robot has a firewall or in situations in which no graphics server can be run.
+
+## How to use
+
+To launch the visualizer, just run the `termviz` executable.
+
+The program looks for a configuration file named `termviz.yml` in `~/.config/termviz/` first, then in `/etc/termviz/`. If the file is not found, it prompts the user to create a default one in `~/.config/termviz/termviz.yml`. Alternatively, it is possible to pass a configuration file directly to the executable: `termviz <myconfig>.yml`.
+
+The program requires a running ROS master and an available TF between the robot frame (`base_link` by default) and a static frame (`map` by default).
+
+Once the visualizer starts, every configured topic, as well as the robot footprint (obtained from the ROS parameter `/footprint`) and frame, should be visible in the viewport. Pressing `h` shows the help screen, which will describe the current mode and the keymap relative to the current mode. The mode can be switched using the number keys and the help screen will update accordingly.
+
+### Send pose mode
+
+The mode allows to publish a `geometry_msgs::PoseWithCovarianceStamped` message on a topic. The desired pose can be selected by moving the outline of the robot in the map. Confirming the operation (`Enter` by default) publishes the pose on the topic specified under `send_pose_topic` in the configuration file.
+
+### Teleoperate mode
+
+The mode allows to teleoperate the robot by sending `geometry_msgs::Twist` messages on the specified topic (`cmd_vel` by default). The messages are continuosly sent. Any unmapped key switches the sent messages to 0, i.e., stops the robot.
+Settings can be found under `teleop` in the configuration file.
+
+### Image mode
+
+This mode allows to visualize images received on the topics specified under `image_topics` in the configuration file.
+
+## Supported message types
+
+- geometry_msgs::PoseArray
+- geometry_msgs::PoseStamped
+- nav_msgs::OccupancyGrid
+- nav_msgs::Path
+- sensor_msgs::Image
+- sensor_msgs::LaserScan
+- sensor_msgs::PointCloud2
+- visualization_msgs::Marker
+- visualization_msgs::MarkerArray
+
+## Default config
+
+Here is the commented default config file:
```

Comment:
should we not just link the file in config? then we are safe against changes

Reply:
Do you know if it's possible to autogenerate a config file with the comments directly? That would be the best solution. I personally like the comments in the README, because if you quickly need to modify the config you immediately know where to find a reference, but I'm open to suggestions...